source/fe/fe_hermite.cc

```diff
@@ -0,0 +1,840 @@
+  {
+    return Utilities::invert_permutation(
+      hermite_hierarchic_to_lexicographic_numbering<dim>(regularity));
+  }
+
+
+
+  template <int dim>
+  inline std::vector<unsigned int>
+  hermite_face_lexicographic_to_hierarchic_numbering(
+    const unsigned int regularity)
+  {
+    return (dim > 1) ? hermite_lexicographic_to_hierarchic_numbering<dim - 1>(
+                         regularity) :
+                       std::vector<unsigned int>();
+  }
+
+
+
+  template <int dim>
+  TensorProductPolynomials<dim>
+  get_hermite_polynomials(const unsigned int regularity)
+  {
+    TensorProductPolynomials<dim> poly_space(
+      Polynomials::HermitePoly::generate_complete_basis(regularity));
+
+    std::vector<unsigned int> renumber =
+      internal::hermite_hierarchic_to_lexicographic_numbering<dim>(regularity);
+    poly_space.set_numbering(renumber);
+
+    return poly_space;
+  }
+
+
+
+  static inline unsigned int
+  binomial(const unsigned int n, const unsigned int i)
+  {
+    unsigned int C = 1, k = 1;
+
+    for (unsigned int j = n; j > i; --j)
+      {
+        C *= j;
+        C /= k++;
+      }
+
+    return C;
+  }
+
+
+
+  template <int xdim, int xspacedim = xdim, typename xNumber = double>
+  class Rescaler
+  {
+  public:
+    void
+    rescale_fe_hermite_values(
+      const FE_Hermite<xdim, xspacedim> &                        fe_herm,
+      const typename Mapping<xdim, xspacedim>::InternalDataBase &mapping_data,
+      Table<2, xNumber> &                                        value_list);
+
+
+
+    template <int spacedim, typename Number>
+    void
+    rescale_fe_hermite_values(
+      Rescaler<1, spacedim, Number> & /*rescaler*/,
+      const FE_Hermite<1, spacedim> &                        fe_herm,
+      const typename Mapping<1, spacedim>::InternalDataBase &mapping_data,
+      Table<2, Number> &                                     value_list)
+    {
+      unsigned int n_q_points;
+      double       cell_extent = 1.0;
+
+      // Check mapping_data is associated with a compatible mapping class
+      if (dynamic_cast<const typename MappingHermite<1, spacedim>::InternalData
+                         *>(&mapping_data) != nullptr)
+        {
+          const typename MappingHermite<1, spacedim>::InternalData *data =
+            dynamic_cast<
+              const typename MappingHermite<1, spacedim>::InternalData *>(
+              &mapping_data);
+          n_q_points  = data->quadrature_points.size();
+          cell_extent = data->cell_extents[0];
+        }
+      else if (dynamic_cast<const typename MappingCartesian<1>::InternalData *>(
+                 &mapping_data) != nullptr)
+        {
+          const typename MappingCartesian<1>::InternalData *data =
+            dynamic_cast<const typename MappingCartesian<1>::InternalData *>(
+              &mapping_data);
+          n_q_points  = data->quadrature_points.size();
+          cell_extent = data->cell_extents[0];
+        }

```

Comment:
I guess the two cases are identical. If yes, I would remove one and add a case if there is indeed a new mapping class.